src/azguard.ts

```diff
@@ -103,9 +105,36 @@ class ShieldSwapAzguardProvider implements TypedEip1193Provider {
       if (!account) {
         throw new Error("Unauthorized account");
       }
+      const chain = account.substring(0, account.lastIndexOf(":"));
+
+      const operations = [];
+
+      if (request.registerContracts) {
+        operations.push(
+          ...request.registerContracts.map((x) => ({
+            kind: "register_contract",
+            chain,
+            address: x.address,
+            instance: x.instance ? {...x.instance, address: x.address} : undefined,
+            artifact: x.artifact ? JSON.parse(x.artifact) : undefined,
```

Comment:
artifact is serialised/deserialised using aztec.js specific `jsonStringify` and `jsonParseWithSchema`

https://github.com/olehmisar/aztec-wallet-sdk/blob/e6a22eb9986dcbe003bd11fb6f638649de9ecd2e/src/serde.ts#L147-L160

Are you sure regular `JSON.parse` will work here?

Reply:
`jsonParseWithSchema` uses `JSON.parse` under the hood as well, so nothing should break